src/mscorlib/shared/System/Memory.cs

```diff
@@ -0,0 +1,313 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using EditorBrowsableAttribute = System.ComponentModel.EditorBrowsableAttribute;
+using EditorBrowsableState = System.ComponentModel.EditorBrowsableState;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    public struct Memory<T>
+    {
+        // The highest order bit of _index is used to discern whether _arrayOrOwnedMemory is an array or an owned memory
+        // if (_index >> 31) == 1, object _arrayOrOwnedMemory is an OwnedMemory<T>
+        // else, object _arrayOrOwnedMemory is a T[]
+        readonly object _arrayOrOwnedMemory;
+        readonly int _index;
+        readonly int _length;
+
+        private const int bitMask = 0x7FFFFFFF;
```

Comment:
Nit: BitMask

Reply:
Actually, in seeing the usage of it in the code, it'd be nice if the name specified what it was a bit mask for, e.g. `RemoveOwnedFlagBitMask` or something like that.